wicker/core/datasets.py

```diff
@@ -35,17 +94,127 @@ def __len__(self) -> int:
         """Returns the length of the dataset/version/partition"""
         pass
 
-    @abc.abstractmethod
+    @property
     def schema(self) -> DatasetSchema:
         """Return the schema of the dataset."""
-        pass
+        schema_data = None
+        schema_path = self._path_factory._get_dataset_schema_path(self._dataset_id)
+        local_path = schema_path
+        if self._column_bytes_file_cache is not None and self._local_cache_path_prefix is not None:
+            local_path = self._storage.fetch_file(
```

Comment:
It looks to me that the `self._storage.fetch_file` call relies only on the `self._local_cache_path_prefix is not None` check above and not the `self._column_bytes_file_cache is not None` part of the check. If this is true, let's rewrite the if-condition in the line above.

Reply:
@isaak-willett in the original function at https://github.com/woven-planet/wicker/blob/main/wicker/core/datasets.py#L105, there is a check to see if the schema has already been cached as `_self.schema`; can we restore this part of the function?

Reply:
@convexquad I added the property decorator so it does the same thing through that decorator. Are you okay with me leaving the local cache prefix path here because now the `_column_bytest_file_reader` is not going be none ever?

Reply:
Oh I totally am! I forgot about that, damnit.

Reply:
Fixed, used `cached_property`